File: Scripts/pageobjects/NewArticle_po.py
from BasePage import BasePage


class NewArticle(BasePage):
    et_title = "//input[@placeholder='Article Title']"
    et_description = '//input[@placeholder="What\'s this article about?"]'
    et_article_text = "//textarea[@placeholder='Write your article (in markdown)']"
    et_tags = "//input[@placeholder='Enter tags']"
    bt_publish = "//button[@class='btn btn-lg pull-xs-right btn-primary']"

    def input_title(self, title):
        self._driver.find_elements_by_xpath(self.et_title)[0].send_keys(title)

    def input_description(self, description):
        self._driver.find_elements_by_xpath(self.et_description)[0].send_keys(description)

    def input_article_text(self, text):
        self._driver.find_elements_by_xpath(self.et_article_text)[0].send_keys(text)

    def input_tags(self, tags):
        self._driver.find_elements_by_xpath(self.et_tags)[0].send_keys(tags)

    def publish(self):
        self._driver.find_elements_by_xpath(self.bt_publish)[0].click()

File: angularjs-realworld-example-app-master/tests/PageObjects/NewArticle.po.js
var NewArticle = function(){
    this.title = element.all(by.xpath("//input[@placeholder='Article Title']")).first();
    this.description = element.all(by.xpath('//input[@placeholder="What\'s this article about?"]')).first();
    this.article_text = element.all(by.xpath("//textarea[@placeholder='Write your article (in markdown)']")).first();
    this.tags = element.all(by.xpath("//input[@placeholder='Enter tags']")).first();
    this.publish = element(by.xpath("//button[@class='btn btn-lg pull-xs-right btn-primary']")).first();
};

NewArticle.prototype.input_title = function(title) {
    return this.title.sendKeys(title);
};

NewArticle.prototype.input_description = function(description) {
    return this.description.sendKeys(description);
};

NewArticle.prototype.input_article_text = function(article_text) {
    return this.article_text.sendKeys(article_text);
};

NewArticle.prototype.input_tags = function(tags) {
    return this.tags.sendKeys(tags);
};

NewArticle.prototype.publish = function() {
    this.publish.click();
};

module.exports = NewArticle;